test/jdk/java/nio/channels/FileChannel/BlockDeviceSize.java

```diff
@@ -34,31 +34,40 @@
 import java.nio.channels.FileChannel;
 import java.nio.file.AccessDeniedException;
 import java.nio.file.NoSuchFileException;
+import java.util.List;
+
 import static java.nio.file.StandardOpenOption.*;
 
 
 public class BlockDeviceSize {
-    private static final String BLK_FNAME = "/dev/sda1";
-    private static final Path BLK_PATH = Paths.get(BLK_FNAME);
+    private static final List<String> BLK_FNAMES = List.of("/dev/sda1", "/dev/nvme0n1") ;
```

Comment:
Is this list sufficient? Has any kind of survey been performed to determine whether anything else should be in the list?

Reply:
The devices included are only the two most popular block devices in Linux. I'd tend to say that having those two should provide enough coverage, but I'm open to including others. For reference, this [link](https://github.com/brianmcgillion/udev/blob/master/rules/rules.d/60-persistent-storage.rules) gives an idea of the rules that udev uses to name devices and hints for example at the fact that I've not included `/dev/mmcblk0` that is popular, for example in the raspberry pis.

Reply:
I'm not familiar enough with which devices need to be covered. If there are more devices of non-negligible interest, however, and adding them would neither destabilize the test nor substantially increase the test's running time, then I would not hesitate to add a few. On the other if you think this is enough, I cannot disagree.

Reply:
I've thought about it and in the end I've also added `/dev/xvda1`. That should allow running the test in virtualized environments, what should increase significantly the coverage. I believe those three devices should be a very representative set of the environments where this test will be run.